### PR TITLE
Documentation: About vald

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

Create `Aboud Vald` document. The table of content is as follow.

```
- [What is Vald?](#what-is-vald)
  - [What Vald can do?](#what-vald-can-do)
  - [Use cases](#use-cases)
- [Why Vald?](#why-vald)
- [How does Vald work?](#how-does-vald-work)
- [Try Vald](#try-vald)
```

`What Vald can do?` and `Use cases` section is copied from `README.md`.

The remaining task/question is as follow:

- [ ] How to keep the consistency of the copied contents?

### Environment:

- Golang Version: 1.14
- Docker Version: 19.03.5
- Kubernetes Version: 1.17.3
- NGT Version: 1.9.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/overview/about.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- [How does Vald work?](#how-does-vald-work)
+- [Try Vald](#try-vald)
+
+### What is Vald?
+
+Please refer to [this page](https://github.com/vdaas/vald#what-is-vald) for more details.
+
+#### Use cases
+
+Vald support similarity searching.
```
Comment (hlts2) [inline docs/overview/about.md:17]:
```suggestion
Vald supports similarity searching.
```

Comment (hlts2) [inline docs/overview/about.md:58]:
```suggestion
Vald supports similarity searching.
```

File: docs/overview/about.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- Speech recognition
+- Everything you can vectorize :)
+
+### Why Vald?
+
+Vald is based on Kubernetes and Cloud-Native architecture, which means Vald is highly scalable. You can easily scale Vald by changing Vald's configuration.
+
+Vald uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors by default, but users can switch to another vector searching engine in Vald to support the best performance for your use case.
+
+Also, Vald support auto-healing, to reduce running and maintenance cost. Vald implements the backup mechanism to support diaster recovery. Whenever Vald Agent is down, the new Vald Agent instance will be created automatically and the data will be recovered automatically.
```
Comment (hlts2) [inline docs/overview/about.md:70]:
```suggestion
Also, Vald supports auto-healing, to reduce running and maintenance cost. Vald implements the backup mechanism to support diaster recovery. Whenever Vald Agent is down, the new Vald Agent instance will be created automatically and the data will be recovered automatically.
```

File: docs/overview/about.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### How does Vald work?
+
+Vald implements its custom resource and custom controller to integrate with Kubernetes; you can take all the benefits from Kubernetes.
```
Comment (hlts2) [inline docs/overview/about.md:74]:
```suggestion
Vald implements its custom resource and custom controller to integrate with Kubernetes.
you can take all the benefits from Kubernetes.
```

File: docs/overview/about.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- Multi language supported
+    - Golang, Java, Nodejs and python is supported.
+
+#### Use cases
+
+Vald support similarity searching.
+
+- Related image search
+- Speech recognition
+- Everything you can vectorize :)
```
Comment (hlts2) [inline docs/overview/about.md:62]: Could you please mean the following word ? 🙏 
```
:)
```

File: docs/overview/about.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+Vald is based on Kubernetes and Cloud-Native architecture, which means Vald is highly scalable. You can easily scale Vald by changing Vald's configuration.
+
+Vald uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors by default, but users can switch to another vector searching engine in Vald to support the best performance for your use case.
+
+Also, Vald support auto-healing, to reduce running and maintenance cost. Vald implements the backup mechanism to support diaster recovery. Whenever Vald Agent is down, the new Vald Agent instance will be created automatically and the data will be recovered automatically.
+
+### How does Vald work?
+
+Vald implements its custom resource and custom controller to integrate with Kubernetes; you can take all the benefits from Kubernetes.
```
Comment (kevindiu) [inline docs/overview/about.md:74]:
```suggestion
Vald implements its custom resource and custom controller to integrate with Kubernetes.
You can take all the benefits from Kubernetes.
```

File: docs/overview/about.md (hunk 4 above)
Comment (kevindiu) [reply docs/overview/about.md:62]: it has no meaning XD

Comment (hlts2) [reply docs/overview/about.md:62]: ok

File: docs/overview/about.md (hunk 3 above)
Comment (hlts2) [reply docs/overview/about.md]: Sorry. Coud you please teach me this word.
I think taht period is better.:thinking:
```
;
```

File: docs/overview/about.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+Vald supports similarity searching.
+
+- Related image search
+- Speech recognition
+- Everything you can vectorize :)
+
+### Why Vald?
+
+Vald is based on Kubernetes and Cloud-Native architecture, which means Vald is highly scalable. You can easily scale Vald by changing Vald's configuration.
```
Comment (vankichi) [inline docs/overview/about.md:66]:
```suggestion
Vald is based on Kubernetes and Cloud-Native architecture, which means Vald is highly scalable.
You can easily scale Vald by changing Vald's configuration.
```

File: docs/overview/about.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+Vald uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors by default, but users can switch to another vector searching engine in Vald to support the best performance for your use case.
+
+Also, Vald supports auto-healing, to reduce running and maintenance cost. Vald implements the backup mechanism to support diaster recovery. Whenever Vald Agent is down, the new Vald Agent instance will be created automatically and the data will be recovered automatically.
```
Comment (vankichi) [inline docs/overview/about.md:70]:
```suggestion
Also, Vald supports auto-healing, to reduce running and maintenance costs. Vald implements the backup mechanism to support disaster recovery.
Whenever one of the Vald Agent instances is down, the new Vald Agent instance will be created automatically and the data will be recovered automatically.
```

**Comment by kevindiu** [comment]
I think about it by myself. I think that it is better to keep it as simple as possible in this document as it is the introduction page for Vald. 
@vankichi  If you think this PR is LGTMed please process to author review :)

File: docs/overview/about.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# About Vald <!-- omit in toc -->
+
```
Comment (vankichi) [inline docs/overview/about.md:2]: Could you please describe this document's abstract here ?

Comment (kevindiu) [reply docs/overview/about.md:2]: How about this?
```suggestion

This document gives an overview of what is Vald and what you can do with Vald. 
```

Comment (vankichi) [reply docs/overview/about.md:2]: LGTM

File: docs/overview/about.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+This document gives an overview of what is Vald and what you can do with Vald. 
+## Table of Contents <!-- omit in toc -->
```
Comment (vankichi) [inline docs/overview/about.md:4]:
```suggestion

## Table of Contents <!-- omit in toc -->
```

File: docs/overview/about.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+### How does Vald work?
+
+Vald implements its custom resource and custom controller to integrate with Kubernetes.
+You can take all the benefits from Kubernetes.
+
+Please refer to the [architecture overview](./architecture.md) for more details about the architecture and how each component in Vald works together.
+
+### Try Vald
+
+Please refer to [Get Started](./get-started.md) to try Vald :)
```
Comment (vankichi) [inline docs/overview/about.md:84]:
```suggestion
Please refer to [Get Started](../get-started/get-started.md) to try Vald :)
```

Comment (kevindiu) [inline docs/overview/about.md:84]:
```suggestion
Please refer to [Get Started](../user/get-started.md) to try Vald :)
```

File: docs/overview/about.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+It uses the fastest ANN Algorithm NGT to search neighbors.
+Vald has automatic vector indexing and index backup, and horizontal scaling which made for searching from billions of feature vector data.
+Vald is easy to use, feature-rich and highly customizable as you needed.
+
+#### What Vald can do?
+
+<!--  copied -->
+
+- Asynchronize Auto Indexing
+    - Usually the graph requires locking during indexing, which cause stop-the-world. But Vald uses distributed index graph so it continues to work during indexing.
```
Comment (hlts2) [inline docs/overview/about.md:28]:
```suggestion
    - Usually the graph requires locking during indexing, which causes stop-the-world. But Vald uses a distributed index graph so it continues to work during indexing.
```

File: docs/overview/about.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+- Customizable Ingress/Egress Filtering
+    - Vald implements it's own highly customizable Ingress/Egress filter.
+    - Which can be configured to fit the gRPC interface.
+        - Ingress Filter: Ability to Vectorize through filter on request.
+        - Egress Filter: rerank or filter the searching result with your own algorithm.
+
+- Cloud-native based vector searching engine
+    - Horizontal scalable on memory and cpu for your demand.
```
Comment (hlts2) [inline docs/overview/about.md:37]:
```suggestion
    - Horizontal scalable on memory and CPU for your demand.
```

File: docs/overview/about.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+- Auto Indexing Backup
+    - Vald has auto index backup feature using MySQL + Redis or Cassandra which enables disaster recovery.
```
Comment (hlts2) [inline docs/overview/about.md:40]:
```suggestion
    - Vald has an auto index backup feature using MySQL + Redis or Cassandra which enables disaster recovery.
```

File: docs/overview/about.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+- Distributed Indexing
+    - Vald distribute vector index to multiple agent, each agent stores different index.
```
Comment (hlts2) [inline docs/overview/about.md:43]:
```suggestion
    - Vald distribute vector index to multiple agents, each agent stores different index.
```

File: docs/overview/about.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+- Index Replication
+    - Vald stores each index in multiple agents which enables index replicas.
+    - Automatically rebalance the replica when some Vald agent goes down.
+
+- Easy to use
+    - Vald can be easily installed in a few steps.
+
+- Highly customizable
+    - You can configure the number of vector dimension, the number of replica and etc.
```
Comment (hlts2) [inline docs/overview/about.md:53]:
```suggestion
    - You can configure the number of vector dimensions, the number of replica and etc.
```

File: docs/overview/about.md [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+
+- Multi language supported
+    - Golang, Java, Nodejs and python is supported.
```
Comment (hlts2) [inline docs/overview/about.md:56]:
```suggestion
    - Golang, Java, Nodejs, and python is supported.
```

File: docs/overview/about.md (hunk 11 above)
Comment (vankichi) [reply docs/overview/about.md:28]: I think below is better.
```suggestion
    - Usually the graph requires locking during indexing, which causes stop-the-world. But Vald uses distributed index graphs so it continues to work during indexing.
```

File: docs/overview/about.md (hunk 16 above)
Comment (kevindiu) [inline docs/overview/about.md:56]:
```suggestion
    - Golang, Java, Nodejs, and python are supported.
```

**Comment by hlts2** [comment]
LGTM

**Comment by hlts2** [comment]
LGTM

File: README.md [hunk 17]
```diff
@@ -19,19 +19,25 @@
...
-Vald is designed and implemented based on the Cloud-Native architecture. <br>
-It uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors.<br>
-Vald has automatic vector indexing and index backup, and horizontal scaling which made for searching from billions of feature vector data.<br>
-Vald is easy to use, feature-rich and highly customizable as you needed.<br>
+
+Vald is a highly scalable distributed fast approximate nearest neighbor dense vector search engine.
+
+Vald is designed and implemented based on Cloud-Native architecture.
+
+It uses the fastest ANN Algorithm NGT to search neighbors.
```
Comment (vankichi) [inline README.md:27]:
```suggestion
It uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors.
```

File: docs/overview/about.md [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+- [Try Vald](#try-vald)
+
+### What is Vald?
+
+<!-- copied from README.md-->
+Vald is a highly scalable distributed fast approximate nearest neighbor dense vector search engine.
+
+Vald is designed and implemented based on Cloud-Native architecture.
+
+It uses the fastest ANN Algorithm NGT to search neighbors.
```
Comment (vankichi) [inline docs/overview/about.md:21]:
```suggestion
It uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors.
```

File: README.md [hunk 19]
```diff
@@ -56,10 +62,10 @@ Go to [Get Started](./docs/user/get-started.md) page to try out Vald :)
     - Vald can be easily installed in a few steps.
 
 - Highly customizable
-    - You can configure the number of vector dimension, the number of replica and etc.
+    - You can configure the number of vector dimensions, the number of replica and etc.
 
 - Multi language supported
-    - Golang, Java, Nodejs and python is supported.
+    - Golang, Java, Nodejs, and python are supported.
```
Comment (rinx) [inline README.md:68]: badge
```suggestion
    - Golang, Java, Clojure, Node.js, and Python clients are supported.
    - gRPC APIs can be called by any other languages that supports gRPC.
    - REST APIs are provided.
```

File: docs/overview/about.md [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    - Automatically rebalance the replica when some Vald agent goes down.
+
+- Easy to use
+    - Vald can be easily installed in a few steps.
+
+- Highly customizable
+    - You can configure the number of vector dimensions, the number of replica and etc.
+
+- Multi language supported
+    - Golang, Java, Nodejs, and python are supported.
```
Comment (rinx) [inline docs/overview/about.md:59]: same

File: docs/overview/about.md [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+    - Vald implements it's own highly customizable Ingress/Egress filter.
+    - Which can be configured to fit the gRPC interface.
+        - Ingress Filter: Ability to Vectorize through filter on request.
+        - Egress Filter: rerank or filter the searching result with your own algorithm.
+
+- Cloud-native based vector searching engine
+    - Horizontal scalable on memory and CPU for your demand.
+
+- Auto Indexing Backup
+    - Vald has an auto index backup feature using MySQL + Redis or Cassandra which enables disaster recovery.
```
Comment (rinx) [inline docs/overview/about.md:43]: badge
to tell the truth, Redis is not used for index backup, just for storing meta KV. :rofl:

Comment (rinx) [inline docs/overview/about.md:43]:
```suggestion
    - Vald has a feature to store the backup of the index data using MySQL or Cassandra which enables disaster recovery.
```

File: docs/overview/about.md [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+- Customizable Ingress/Egress Filtering
+    - Vald implements it's own highly customizable Ingress/Egress filter.
+    - Which can be configured to fit the gRPC interface.
+        - Ingress Filter: Ability to Vectorize through filter on request.
+        - Egress Filter: rerank or filter the searching result with your own algorithm.
+
+- Cloud-native based vector searching engine
+    - Horizontal scalable on memory and CPU for your demand.
+
+- Auto Indexing Backup
```
Comment (rinx) [inline docs/overview/about.md:42]:
```suggestion
- Auto Backup for Index data
```

File: README.md [hunk 23]
```diff
@@ -40,13 +46,13 @@ Go to [Get Started](./docs/user/get-started.md) page to try out Vald :)
         - Egress Filter: rerank or filter the searching result with your own algorithm.
 
 - Cloud-native based vector searching engine
-    - Horizontal scalable on memory and cpu for your demand.
+    - Horizontal scalable on memory and CPU for your demand.
 
 - Auto Indexing Backup
-    - Vald has auto index backup feature using MySQL + Redis or Cassandra which enables disaster recovery.
+    - Vald has an auto index backup feature using MySQL + Redis or Cassandra which enables disaster recovery.
```
Comment (rinx) [inline README.md:52]: same

File: README.md (hunk 19 above)
Comment (kevindiu) [reply README.md:68]: maybe this is better
```suggestion
    - Golang, Java, Clojure, Node.js, and Python client library is supported.
    - gRPC APIs can be triggered by any programming languages which support gRPC.
    - REST API is also supported.```

File: README.md (hunk 23 above)
Comment (kevindiu) [reply README.md:52]:
```suggestion
    - Vald has a feature to store the backup of the index data using MySQL or Cassandra which enables disaster recovery.
```

File: README.md [hunk 24]
```diff
@@ -40,13 +46,13 @@ Go to [Get Started](./docs/user/get-started.md) page to try out Vald :)
         - Egress Filter: rerank or filter the searching result with your own algorithm.
 
 - Cloud-native based vector searching engine
-    - Horizontal scalable on memory and cpu for your demand.
+    - Horizontal scalable on memory and CPU for your demand.
 
 - Auto Indexing Backup
```
Comment (kevindiu) [inline README.md:51]:
```suggestion
- Auto Backup for Index data
```

File: README.md [hunk 25]
```diff
@@ -19,19 +19,25 @@
...
+
+Vald is easy to use, feature-rich and highly customizable as you needed.
+
 Go to [Get Started](./docs/user/get-started.md) page to try out Vald :)
 
 (If you are interested in ANN benchmarks, please refer to [the official website](http://ann-benchmarks.com/).)<br>
 
 ### Main Features
 
 - Asynchronize Auto Indexing
```
Comment (rinx) [inline README.md:39]: maybe
```suggestion
- Asynchronous Auto Indexing
```

File: docs/overview/about.md [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+It uses the fastest ANN Algorithm [NGT](https://github.com/yahoojapan/NGT) to search neighbors.
+
+Vald has automatic vector indexing and index backup, and horizontal scaling which made for searching from billions of feature vector data.
+
+Vald is easy to use, feature-rich and highly customizable as you needed.
+
+#### What Vald can do?
+
+<!-- copied from README.md-->
+- Asynchronize Auto Indexing
```
Comment (kevindiu) [inline docs/overview/about.md:30]:
```suggestion
- Asynchronous Auto Indexing
```

**Pull request merged**